src/k3s/mod.rs

```diff
@@ -0,0 +1,160 @@
+use std::collections::HashMap;
+use std::io;
+use std::io::ErrorKind;
+use std::path::{Path, PathBuf};
+
+use testcontainers::core::{Mount, WaitFor};
+use testcontainers::{Image, ImageArgs};
+
+const NAME: &str = "rancher/k3s";
+const TAG: &str = "v1.28.8-k3s1";
+pub const TRAEFIK_HTTP: u16 = 80;
+pub const KUBE_SECURE_PORT: u16 = 6443;
+pub const RANCHER_WEBHOOK_PORT: u16 = 8443;
+
+#[derive(Debug, Default, Clone)]
+pub struct K3s {
+    env_vars: HashMap<String, String>,
+    conf_mount: Option<Mount>,
+}
+
+#[derive(Default, Debug, Clone)]
+pub struct K3sArgs;
+
+impl ImageArgs for K3sArgs {
+    fn into_iterator(self) -> Box<dyn Iterator<Item = String>> {
+        Box::new([String::from("server"), String::from("--snapshotter=native")].into_iter())
```

Comment:
Don't we need to expose an ability to specify/override `snapshotter`?

Reply:
Sure, I changed to keep native as default (I believe it is the most useful for test purposes) but now it can be customized via args field.